R/map_df.R

```diff
@@ -250,15 +249,50 @@ map_df <- function(trt_md,
         lapply(ref, function(x) stats::setNames(x, names(ref[[1]])))
       )
     )
-    
     # match trt and ref
     matchTrtRef <- matches(trt, ref, list = FALSE, all.y = FALSE)
     matchTrtRef[["x"]] <- trtNames[matchTrtRef[["x"]]]
     matchTrtRef[["y"]] <- refNames[matchTrtRef[["y"]]]
-    
     out <- split(matchTrtRef[["y"]], matchTrtRef[["x"]])
+    # remove cases where the match is missing in the reference
+    out <- lapply(out, function(x) x[!is.na(x)])
     out
   } else {
     out
   }
 }
+
+
+#' Identify untreated rows based on Drug treatment alone
+#' 
+#' @param mat_elem input data frame
+#'
+#' @details
+#' Using the given rownames, map the untreated conditions
+#' 
+#' @return list
+#' 
+map_untreated <- function(mat_elem) {
+  # TODO: avoid recycling code of map_references above
```

Comment:
Could it be described as jira task?

Reply:
Done. [GDR-2332](https://jira.gene.com/jira/browse/GDR-2332)